src/fair/simulation.py

```diff
@@ -98,3 +98,74 @@ def __init__(
         ]
 
         super().__init__(constraints, memoize)
+
+
+class SubStudent(SimulatedAgent):
+    """Randomly generate a Renaissance man student
```

Comment:
I created a static method to build a sub student without creating a whole new class. The RenaissanceMan now receives a `sample` bool parameter set to True by default. If True, then we sample a new RenaissanceMan as always. If set to False, instead of using the random seed to build a new student, it simply creates a new student from the existing values given (preferred_topics, preferred_courses, quantities and total_courses, inherited from the original student). RenaissanceMan now takes these parameters which are set to None by default, and when creating a sub_student, we pass these parameters from the original student and avoid the sampling. Not sure if this is the cleanest way to do it.